resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ZipFileTransformer.java

```diff
@@ -0,0 +1,113 @@
+import org.eclipse.core.filesystem.IFileStore;
+import org.eclipse.core.filesystem.URIUtil;
+import org.eclipse.core.filesystem.ZipFileUtil;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+
+/**
+ * Utility class for opening and closing zip files.
+ *
+ * @since 3.21
+ */
+public class ZipFileTransformer {
+
+	/**
+	 * Closes an opened zip file represented as a linked folder in the workspace.
+	 * After closing, the zip file in its file state is shown in the workspace.
+	 *
+	 * This method can only be called when the zip file is local. Otherwise a
+	 * CoreException is thrown.
+	 *
+	 * @param folder The folder representing the zip file to close.
+	 *
+	 */
+	public static void closeZipFile(IFolder folder) throws URISyntaxException, CoreException {
+		URI zipURI = new URI(folder.getLocationURI().getQuery());
+		IFileStore parentStore = EFS.getStore(folder.getParent().getLocationURI());
+		URI childURI = parentStore.getChild(folder.getName()).toURI();
+		if (URIUtil.equals(zipURI, childURI)) {
+			folder.delete(IResource.CLOSE_ZIP_FILE, null);
+			folder.getProject().refreshLocal(IResource.DEPTH_INFINITE, null);
+		} else {
+			throw new CoreException(new Status(IStatus.ERROR, ResourcesPlugin.PI_RESOURCES,
+					"Closing of Zip File " + folder.getName() //$NON-NLS-1$
+							+ " failed because the Zip File is not local.")); //$NON-NLS-1$
+		}
+	}
+
+	/**
+	 * Opens a zip file represented by a file into a linked folder. The zip file
+	 * will not be extracted in this process. In the opened state the linked folder
+	 * allows reading and manipulating the children of the zip file in the workspace
+	 * and on the filesystem. If the folder has no children after opening, then it
+	 * is closed immediately after. In this case it could be that there should be no
+	 * children, so there is no need for opening or an error occured that prevented
+	 * the children from being loaded.
+	 *
+	 * This method prevents opening linked zip files. zip files must be local to be
+	 * opened. Otherwise a CoreException is thrown.
+	 *
+	 * @param file    The file representing the zip file to open.
+	 * @param monitor monitor indicating the completion progress
+	 *
+	 */
+	public static void openZipFile(IFile file, IProgressMonitor monitor, boolean backgroundRefresh)
+			throws URISyntaxException, CoreException {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, 20);
+		try (InputStream fis = file.getContents()) {
+			ZipFileUtil.checkFileForZipHeader(fis);
+			// Additional operations can continue here if header is correct
+		} catch (IOException e) {
+			// If the header is incorrect or there's an IO error, handle gracefully
+			throw new CoreException(new Status(IStatus.ERROR, "your.plugin.id", //$NON-NLS-1$
```

Comment:
Is `"your.plugin.id"` the correct id?

Reply:
changed in fd7727f12d9fd0c1a90fab8e6bd27c246a1308bb